docs/guides/migrations/guide.rst

```diff
@@ -0,0 +1,1298 @@
+``edgedb migration extract``. This command will retrieve the migration(s) 
+created using DDL and assign each of them a proper file name and hash 
+inside the ``/migrations`` folder, effectively giving them a proper 
+position inside the migration flow.
+
+Note that at this point your ``.esdl`` schema will still not match 
+the database schema, so if you were to type ``edgedb migration create`` 
+the CLI would then ask you if you want to drop the type that you just 
+created - because it doesn't exist inside there. So be sure to change 
+your schema to match the schema inside the database that you have 
+manually changed via DDL. If in doubt, use ``describe schema as sdl`` 
+to compare or use ``edgedb migration create`` and check the output. 
+If the CLI is asking you if you want to drop a type, that means that 
+you forgot to add it to the schema inside your ``.esdl`` file(s).
+
+
+Multiple migrations to keep data 
+================================
+
+Let's say that we have a simple ``User`` type with a ``friends`` link 
+to other ``User`` objects.
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+          name: str;
+          multi friends: User;
+      }
+    }
+
+First let's insert three ``User`` objects, followed by an update to 
+make each ``User`` friends with all of the others:
+
+.. code-block:: edgeql-repl
+
+    db> insert User {
+    ... name := 'User 1'
+    ... };
+    {default::User {id: d44a19bc-8bc1-11ee-8f28-47d7ec5238fe}}
+    db> insert User {
+    ... name := 'User 2'
+    ... };
+    {default::User {id: d5f941c0-8bc1-11ee-8f28-b3f56009a7b0}}
+    db> insert User {
+    ... name := 'User 3'
+    ... };
+    {default::User {id: d79cb03e-8bc1-11ee-8f28-43fe3f68004c}}
+    db> update User set { 
+    ...    friends := (select detached User filter User.name != .name)
+    ...  };
+
+Now what happens if we now want to change ``multi friends`` to an 
+``array<str>``? If we were simply changing a scalar property to another 
+property it would be easy, because EdgeDB would prompt us for a conversion 
+expression. But a change from a link to a property is different:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+          name: str;
+          multi friends: array<str>;
+      }
+    }
+
+Doing a migration as such will just drop the ``friends`` link (along 
+with its data) and create a new ``friends`` property - without any 
+data at all.
+
+To solve this problem, we can do two migrations instead of one. First 
+we will keep the ``friends`` link, while adding a new property called 
+``friend_names``:
+
+.. code-block:: sdl
+
+    module default {
+      type User {
+        name: str;
+        multi friends: User;
+        friend_names: array<str>;
+      }
+    }
+
+The CLI will simply ask us if we created a property called ``friend_names``. 
+We haven't applied the migration yet, so we might as well put the 
+data inside the same migration. A simple update will do the job!
```

Comment:
We don't say how the reader should do this, and I think it would be better if we did. We should probably recommend `edgedb migration edit` for this to make sure the hash gets updated.

Reply:
https://github.com/edgedb/edgedb/pull/6656/commits/f4cea9899aa0da77125e002e770d8a9f3bd15f25